src/snowflake/snowpark/session.py

```diff
@@ -792,7 +793,9 @@ def query_tag(self) -> Optional[str]:
     @query_tag.setter
     def query_tag(self, tag: str) -> None:
         if tag:
-            self._conn.run_query(f"alter session set query_tag = '{tag}'")
+            self._conn.run_query(
+                f"alter session set query_tag = {QUESTION_MARK}", params=(tag,)
```

Comment:
Nit: A straight `?` is fine here and even more readable.

Reply:
`QUESTION_MARK` already exists and is used by batch insertion.